user/test/com/google/gwt/emultest/java10/util/OptionalLongTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.gwt.emultest.java10.util;
+
+import com.google.gwt.emultest.java.util.EmulTestBase;
+
+import java.util.NoSuchElementException;
+import java.util.OptionalLong;
+
+/**
+ * Tests for java.util.OptionalLong Java 10 API emulation.
+ */
+public class OptionalLongTest extends EmulTestBase {
+  public void testOrElseThrow() {
+    try {
+      OptionalLong.empty().orElseThrow();
+      fail("Expected NoSuchElementException from empty Optional: orElseThrow");
+    } catch (NoSuchElementException ignore) {
+      // expected
+    }
+
+    long value = OptionalLong.of(10).orElseThrow();
```

Comment:
Use `10L` here and in the following line. Makes it even more explicit and forces `long` in the method signature of `of()`.